tests/relational/test_relational_data_with_json.py

```diff
@@ -192,47 +220,74 @@ def test_invented_json_column_names(documents, bball):
     ]
 
 
-def test_set_some_primary_key_to_none(documents, bball):
+@patch("gretel_trainer.relational.json.make_suffix")
+def test_set_some_primary_key_to_none(mock_make_suffix, documents):
+    # We need to reset the side_effect iterator later in the test, so we manually mock it
+    mock_make_suffix.side_effect = [f"sfx{i}" for i in range(1, 50)]
+
     # The producer table has a single column primary key,
     # so the root invented table has a composite key that includes the source PK and an invented column
     assert documents.get_primary_key("purchases") == ["id"]
-    assert documents.get_primary_key("purchases-sfx") == ["id", "~PRIMARY_KEY_ID~"]
+    assert documents.get_primary_key(purchases_root_invented_table) == [
+        "id",
+        "~PRIMARY_KEY_ID~",
+    ]
 
     # Setting an existing primary key to None puts us in the correct state
     assert len(documents.list_all_tables(Scope.ALL)) == 5
-    original_payments_fks = documents.get_foreign_keys("payments")
+    original_payments_fks: list[ForeignKey] = documents.get_foreign_keys("payments")
```

Comment:
Just curious if the explicit type hint was needed here?

Reply:
For me the conftest fixtures that yield RelationalData objects are not showing type hints correctly in VS Code.

<img width="862" alt="image" src="https://github.com/gretelai/trainer/assets/26211298/9e87946e-436d-4c08-9086-fa0f223cc2fe">

Probably the better way would be to type hint `documents` to RelationalData in the input params of the test, but maybe there's something we could to to fix this universally for all tests without needing type hints? Not sure.

Reply:
Ooh interesting. So, I used to have type hints on the various RelationalData fixtures in conftest, but I removed them in a [recent PR](https://github.com/gretelai/trainer/pull/130) (here's the [file in the full commit](https://github.com/gretelai/trainer/commit/2b23892796c7a825ebd931b4953110939db5c977#diff-59ef55cb9903ff34cbd594a84ce5791847e83955caeac3361d3f6f9d29024006))—switching from `return` to `yield` was leading to some confusing errors in pyright and I think I just punted on it 😞 

My hypothesis is if we can restore type hints to those fixtures (however they need to be adjusted for the return/yield change), this problem will be solved throughout the rest of the tests.